apps/organizations/apps.py

```diff
@@ -32,6 +32,34 @@ def organization_invitation_created(sender, instance, created, **kwargs):
                 slack_notify(msg)
             except:
                 rollbar.report_exc_info()
+    if created == False:
+        invitation = instance
+
+        if not settings.TEST and htk_setting('HTK_SLACK_NOTIFICATIONS_ENABLED'):
+            if invitation.accepted == True:
+                try:
+                    from htk.utils.notifications import slack_notify
+
+                    msg = invitation.build_notification_message__accepted()
+                    slack_notify(msg)
+                except:
+                    rollbar.report_exc_info()
+            if invitation.accepted == False:
+                try:
+                    from htk.utils.notifications import slack_notify
+
+                    msg = invitation.build_notification_message__declined()
+                    slack_notify(msg)
+                except:
+                    rollbar.report_exc_info()
+            if invitation.accepted == True:
+                try:
+                    from htk.utils.notifications import slack_notify
+
+                    msg = invitation.build_notification_message__resent()
+                    slack_notify(msg)
+                except:
+                    rollbar.report_exc_info()
```

Comment:
- Generally, you should not be re-defining or re-importing the same variables multiple times. The common ones can go outside of the `if/elif/else blocks`, and only the specific code can go into each block.
- For `accepted`, since it is a Null BooleanField, we do have to compare `is True`, `is False`, or `is None` since it can have 3 values instead of the normal 2 (`True`, `False`) for `BooleanField

Here's how I would clean up this function:

```
@disable_for_loaddata
def organization_invitation_updated(sender, instance, created, **kwargs):
    """Signal handler for when a new OrganizationInvitation object is created or updated"""
    invitation = instance
    if not settings.TEST and htk_setting('HTK_SLACK_NOTIFICATIONS_ENABLED'):
        from htk.utils.notifications import slack_notify

        if created:
            msg = invitation.build_notification_message__created()
        else:
            if invitation.accepted is True:
                    msg = invitation.build_notification_message__accepted()
            elif invitation.accepted is False
                    msg = invitation.build_notification_message__declined()
            elif invitation.accepted is None
                    msg = invitation.build_notification_message__resent()
            else:
                # impossible case
                msg = None

        if msg:
            try:
                slack_notify(msg)
            except:
                rollbar.report_exc_info()
```

Then, this cascading if-elif-else can be further cleaned up using another "clever trick" Python pattern:

```
@disable_for_loaddata
def organization_invitation_updated(sender, instance, created, **kwargs):
    """Signal handler for when a new OrganizationInvitation object is created or updated"""
    invitation = instance
    if not settings.TEST and htk_setting('HTK_SLACK_NOTIFICATIONS_ENABLED'):
        from htk.utils.notifications import slack_notify

        if created:
            msg = invitation.build_notification_message__created()
        else:
            INVITATION_ACCEPTANCE_MESSAGE_BUILDERS = {
                True: invitation.build_notification_message__accepted,
                False: invitation.build_notification_message__declined,
                None: invitation.build_notification_message__resent,
            }
            msg_builder = INVITATION_ACCEPTANCE_MESSAGE_BUILDERS.get(invitation.accepted)

        if msg_builder is not None:
            msg = msg_builder()
            try:
                slack_notify(msg)
            except Exception:
                rollbar.report_exc_info()
```

Note on exception handling:
- We want to put as little code as possible in the `try` block, and only wrap the critical section of code which can potentially fail inside of the `try` block
- Always catch *something*. Stylistically, it's good to write `except Exception` rather than just `except`.